libs/core/popover/popover-service/popover.service.ts

```diff
@@ -230,12 +231,29 @@ export class PopoverService extends BasePopoverClass {
         }
     }
 
+    /** Changes background theming when modal */
+    checkModalBackground(): void {
+        if ((!this.closeOnOutsideClick || !this.closeOnEscapeKey) && this.isOpen) {
```

Comment:
Consider reducing the repetition in adding/removing classes by consolidating logic where applicable.
for example: 

`checkModalBackground(): void {
    const bodyClass = 'fd-overlay-active';
    const triggerClass = 'fd-popover__modal';

    if ((!this.closeOnOutsideClick || !this.closeOnEscapeKey) && this.isOpen) {
        this.addClasses(bodyClass, triggerClass);
    } else {
        this.removeClasses(bodyClass, triggerClass);
    }
}`